doc/tutorials/space_shooter/step_6.md

```diff
@@ -0,0 +1,138 @@
+# Enemies and Bullets collision
+
+Right, we real close to a playable game, we have enemies, we have the ability to shoot bullets
+at them! We need now to do something when a bullet hits an enemy.
+
+Flame provides a collision system out of the box, which we will use to implement a logic which when
+a bullet and an enemy comes into contact, both are removed!
+
+First thing we need to do is to let our `FlameGame` that we want collisions between components to
+be checked. In order to do so, simply add the `HasCollisionDetection` missing to the declaration
+of the game class:
+
+```dart
+class SpaceShooterGame extends FlameGame
+    with PanDetector, HasCollisionDetection {
+    // ...
+}
+```
+
+With that, flame now will start to check if components collided with each other. Next we need to
+identify which components can cause collisions.
+
+In our case those are the `Bullet` and `Enemy` components and we need to add a hitbox to them.
+
+A hitbox is nothing more than the dimension which define the area of a component that can hit
+other objects. Flame offers a collection of classes to define a hitbox, the simplest of them is
+the `RectangleHitbox`, which like the name implies will make a rectangular area the component's
+hitbox.
+
+Hitboxes are also components, so, in order to add them to our components we can simply add
+the following line on both the `Bullet` and `Enemy` `onLoad` method:
+
+```dart
+add(RectangleHitbox());
+```
+
+From this point on, Flame will take care of checking the collision between those two components,
+we need now to do something when they came in contact.
+
+We can start that by receiving the collision events in one of the classes. I will choose the
+`Enemy` class to be the one responsible to listening to events, since in theory, there will
+mostly likely have fewer enemies than bullets, so we can less checking if we keep that logic
+on the enemies.
```

Comment:
```suggestion
`Enemy` class to be the one responsible for listening to events, since in theory, there will
mostly likely be fewer enemies than bullets, so we can do less checking if we keep that logic
in the enemy class.
```